llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8909,6 +8914,9 @@ VPlanPtr LoopVectorizationPlanner::tryToBuildVPlanWithVPRecipes(
 
   // Adjust the recipes for any inloop reductions.
   adjustRecipesForReductions(Plan, RecipeBuilder, Range.Start);
+  if (!VPlanTransforms::runPass(
+          VPlanTransforms::handleFMaxReductionsWithoutFastMath, *Plan))
+    return nullptr;
```

Comment:
Not sure why the feature is designed to be in VPlanTransforms?

Reply:
See response here https://github.com/llvm/llvm-project/pull/146711/files/598c59f3c7d565ff7994c816f52c0aaeed138402#r2189438081